tests/test_synced_collection.py

```diff
@@ -345,14 +345,25 @@ def test_keys_with_dots(self, synced_dict):
         with pytest.raises(InvalidKeyError):
             synced_dict['a.b'] = None
 
-    def test_keys_valid_type(self, synced_dict, testdata):
+    def test_keys_str_type(self, synced_dict, testdata):
 
         class MyStr(str):
             pass
-        for key in ('key', MyStr('key'), 0, None, True):
+        for key in ('key', MyStr('key')):
             synced_dict[key] = testdata
-            assert str(key) in synced_dict
-            assert synced_dict[str(key)] == testdata
+            assert key in synced_dict
+            assert synced_dict[key] == testdata
+
+    def test_keys_non_str_valid_type(self, synced_dict, testdata):
```

Comment:
@bdice In #336, we decided not to add backend specific tests in this class. Should I remove this test or not?

Reply:
@vishav1771 If possible, I would only implement this test for the JSON backend (instead of defining it for all backends and skipping on zarr/redis). I would add a comment to the test that says it should be removed in signac 2.0 and mention issue #316.